pkg/kubernetes/secrets/store.go

```
@@ -24,54 +24,56 @@ const (
 }
 
-func NewStore(cfg config.Config, l hclog.Logger) cnabsecrets.Store {
+func NewStore(c *portercontext.Context, cfg PluginConfig) *Store {
+	namespace := cfg.Namespace
 	s := &Store{
-		config:    cfg,
-		logger:    l,
-		hostStore: &host.SecretStore{},
+		Secrets:   make(map[string]map[string]string),
+		hostStore: &cnabhost.SecretStore{},
+		namespace: namespace,
+		logger:    cfg.Logger,
 	}
-
-	return portersecrets.NewSecretStore(s)
+	return s
 }
 
-func (s *Store) Connect() error {
+func (s *Store) connect() error {
 
 	if s.clientSet != nil {
 		return nil
 	}
-
-	clientSet, namespace, err := k8s.GetClientSet(s.config.Namespace, s.logger)
-
+	s.logger.Info(fmt.Sprintf("Store.connect: pre-clientset %s : %s", "namespace", s.namespace))
+	clientSet, namespace, err := k8shelper.GetClientSet(s.namespace)
 	if err != nil {
-		s.logger.Debug(fmt.Sprintf("Failed to get Kubernetes Client Set: %v", err))
 		return err
 	}
+	s.namespace = *namespace
+	s.logger.Info(fmt.Sprintf("Store.connect: post-clientset %s : %s", "namespace", s.namespace))
 
 	s.clientSet = clientSet
-	s.config.Namespace = *namespace
 
 	return nil
 }
 
 func (s *Store) Resolve(keyName string, keyValue string) (string, error) {
+	if err := s.connect(); err != nil {
+		return "", err
+	}
 	if strings.ToLower(keyName) != SecretSourceType {
 		return s.hostStore.Resolve(keyName, keyValue)
 	}
-
+	s.logger.Debug(fmt.Sprintf("Store.Resolve: ns:%s, keyName:%s, keyValue:%s", s.namespace, keyName, keyValue))
 	key := strings.ToLower(keyValue)
 
-	s.logger.Debug(fmt.Sprintf("Looking for key:%s", keyValue))
-	secret, err := s.clientSet.CoreV1().Secrets(s.config.Namespace).Get(context.Background(), key, metav1.GetOptions{})
+	secret, err := s.clientSet.CoreV1().Secrets(s.namespace).Get(context.Background(), key, metav1.GetOptions{})
 	if err != nil {
-		s.logger.Debug(fmt.Sprintf("Failed to Read secrets for key:%s %v", keyValue, err))
 		return "", err
 	}
-
 	return string(secret.Data[SecretDataKey]), nil
```

Let's return an error if the secret doesn't have SecretDataKey. I think that could be a common misunderstanding in how to use the plugin, and any error messages we can surface to help them figure it out will be helpful.

---

@carolynvs added in a custom error for this, just want to double-check wording that it makes sense.